interp/cfg.go

```diff
@@ -87,7 +87,7 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 						}
 					case mapT:
 						n.anc.gen = rangeMap
-						ityp := &itype{cat: ptrT, rtype: reflect.TypeOf((*reflect.MapIter)(nil))}
+						ityp := &itype{cat: valueT, rtype: reflect.TypeOf((*reflect.MapIter)(nil))}
```

Comment:
Can you explain this change?

Reply:
The type here refers to a binary type (vs interpreter defined), from `reflect` package, the right category is valueT, For ptrT category, the data is stored in `val` vs `rtype`. It's for consistency, as `rtype` being filled, the type is usable anyway.

Reply:
OK. Thanks.